src/Kiota.Builder/CodeDOM/CodeClass.cs

```diff
@@ -71,6 +71,20 @@ public void AddInnerClass(params CodeClass[] codeClasses)
             AddMissingParent(codeClasses);
             this.InnerChildElements.AddRange(codeClasses);
         }
+        public CodeClass GetParentClass() {
+            if(StartBlock is Declaration declaration)
+                return declaration.Inherits?.TypeDefinition as CodeClass;
+            else return null;
+        }
+        
+        public CodeClass GetUpperMostInheritanceParent(CodeClass startClassToSkip = null) {
```

Comment:
nit: `Inheritance` is already implied in the `Parent`, so consider shortening the name to `GetUpperMostParent`. I also thought about `GetGreatestGrandparent`, but I am not sure if it is taking the metaphor too far :)

Reply:
I like GetGreatestGrandparent, let's roll with that :)